build/pmm-client-docker/entrypoint.py

```diff
@@ -77,13 +78,20 @@ def main():
 
         agent.wait()
 
-        if status != 0:
+        if status != 0 and not PMM_AGENT_SIDECAR:
             sys.exit(status)
 
-    # use execlp to replace the current process (this entrypoint)
-    # with pmm-agent with inherited environment
-    print('Starting pmm-agent ...', file=sys.stderr)
-    os.execlp('pmm-agent', 'run')
+    # restart pmm-agent if PMM_AGENT_SIDECAR flag is provided
+    while True:
+        # use execlp to replace the current process (this entrypoint)
+        # with pmm-agent with inherited environment.
+        print('Starting pmm-agent ...', file=sys.stderr)
+        os.execlp('pmm-agent', 'run')
```

Comment:
execlp would return here? I am not sure, need to check the spec. Did you test it?
I would think it will replace entry point and exit to init...

Reply:
Yes, I tested it with DBaaS, everything works correctly.

Reply:
Maybe we need to wrap this in `try .. except` and handle this with subprocess?

Could you please add an example of how to check if this works?
My simple code show os.execlp - exit from the parent process and end it.

```
als@mbp qan-api2 % cat  ex.py   
import os

print('------- parent before ---------')
os.execlp('ls', '-l')
print('------- parent after ---------')
als@mbp qan-api2 %           
als@mbp qan-api2 % 
als@mbp qan-api2 % 
als@mbp qan-api2 % 
als@mbp qan-api2 % 
als@mbp qan-api2 % 
als@mbp qan-api2 % 
als@mbp qan-api2 % python3 ex.py
------- parent before ---------
LICENSE                         ca.key                          db.go                           docker-compose.yaml             logs                            mongodb-client.pem              services
Makefile                        ca.srl                          db_test.go                      ex.py                           main.go                         my.cnf                          test_data
README.md                       certificates                    demo.txt                        exporters                       maincover_test.go               mysql.logrotate                 tools
bin                             client.crt                      docker-compose-back.yaml        fixture                         migrations                      server.crt                      utils
ca                              client.key                      docker-compose-dev.yaml         go.mod                          models                          server.key
ca.crt                          client.pem                      docker-compose-pg-tls.yaml      go.sum                          mongodb-CA.pem                  server.pem
als@mbp qan-api2 % 
als@mbp qan-api2 % 

```

As you can see `print('------- parent after ---------')` is not executed when `os.execlp('ls', '-l')` is done.

Reply:
let me check

Reply:
Fixed